### PR TITLE
Upgrade VFS2 to version 2.7.0

### PR DESCRIPTION
**Pull request by fviale**
Use byte arrays for private key authentication

### CODE REVIEW

File: programming-extensions/programming-extension-vfsprovider/src/main/java/org/objectweb/proactive/extensions/vfsprovider/client/ProActiveFileObject.java [hunk 1]
```diff
@@ -850,6 +859,51 @@ private static int getConfiguredNumberOfThreads() {
...
+        @Override
+        public FileObject resolveFile(FileObject baseFile, String name, FileSystemOptions fileSystemOptions)
+                throws FileSystemException {
+            return null;
+        }
+
+        @Override
+        public FileObject resolveFile(String name, FileSystemOptions fileSystemOptions) throws FileSystemException {
+            return null;
+        }
```
Comment (MaelAudren) [inline programming-extensions/programming-extension-vfsprovider/src/main/java/org/objectweb/proactive/extensions/vfsprovider/client/ProActiveFileObject.java:879]: Isn't that better to overload the method resolveFile by adding the missing parameter last in the list of parameters.
i.e
` public FileObject resolveFile(String name, FileSystemOptions fileSystemOptions)`
` public FileObject resolveFile(String name, FileSystemOptions fileSystemOptions, FileObject baseFile)`

Comment (MaelAudren) [reply programming-extensions/programming-extension-vfsprovider/src/main/java/org/objectweb/proactive/extensions/vfsprovider/client/ProActiveFileObject.java:879]: Well I guess those methods come from VfsComponentContext...

**Review by MaelAudren — APPROVED** [review]

**Review by nebilbenmabrouk — APPROVED** [review]
LGTM

File: programming-extensions/programming-extension-vfsprovider/src/main/java/org/objectweb/proactive/extensions/vfsprovider/client/ProActiveFileObject.java (hunk 1 above)
Comment (fviale) [reply programming-extensions/programming-extension-vfsprovider/src/main/java/org/objectweb/proactive/extensions/vfsprovider/client/ProActiveFileObject.java:879]: yes, I have no freedom in the method definition

**Pull request merged**